### ISSUE TITLE
Project Test Case Time Limit Exceeded

### ISSUE THREAD

**Issue by absfsd** [nus-cs2030/2324-s2#773]
![image](https://github.com/nus-cs2030/2324-s2/assets/167889284/14691700-2691-434e-9e1d-a5b163485b44)

is there a testcase i can check this, what are some possible reason this happen at test case 15..

**Comment by sqngwdn** [comment]
i think this means your code has an infinite loop. to rectify, you can add System.out.println(___) in your loop with some kind of variable to check where the loop starts looping infinitely. personally what usually worked for me was printing the event idx or event time.

for instance if you have a while (!pq.isEmpty()) loop there's a chance you might be permanently polling the same event over and over.

**Comment by absfsd** [comment]
> i think this means your code has an infinite loop. to rectify, you can add System.out.println(___) in your loop with some kind of variable to check where the loop starts looping infinitely. personally what usually worked for me was printing the event idx or event time.
> 
> for instance if you have a while (!pq.isEmpty()) loop there's a chance you might be permanently polling the same event over and over.

all the other personal testcases that ive checked theyre all okay as intended but this specific test case 9_15 is hard for me to check .. i gets i can add print statements but in this case idk how

**Comment by TokPinXiang** [comment]
Heyy, i got the same issue, just try again. Sometimes codecrunch have timeout issues. Atb :)

**Comment by jaydenntann77** [comment]
Hi, like the rest have said, there appears to be an infinite loop in your implementation. This could most likely arise from the event being generated after your Wait class, maybe the servers aren't being updated properly or something else.

You can refer to this link https://github.com/nus-cs2030/2122-s2/issues/523 to see if it solves this issue. I encountered the same runtime error issue and this helped me.

**Comment by azzy83** [comment]
Likely an infinite loop bro.

**Comment by starryfieldz** [comment]
Yes, I faced the same issue a few days ago too. It is an infinite loop and likely ur SCC is never serving some customers. You can check your Wait/TransitWait classes, alternatively you can check out these test cases in this post here https://github.com/nus-cs2030/2324-s2/issues/758 and create some println statements to see where exactly is the issue. Hope this helps!

**Comment by khooxr** [comment]
Happened to me too, likely an infinite loop. Can try to check if the events are being removed from your PQ properly.

**Comment by lohkeyyy** [comment]
CS2030 doesnt impose a time limit on our code. No matter how inefficient it is, if it gets the answer, it will be accepted. This likely means your method just does not return an answer ever due to an infinite loop.

**Comment by ezclap02** [comment]
probably an infinite loop. you can try printing strings to check where the infinite loop occurs. for me, it mostly occurred due to not  having the correct event time for Wait and ContinueWaiting events

**Comment by ZHD1987E** [comment]
finals is over you can close this issue
